gcloud/bigtable/row_filters.py

```diff
@@ -119,7 +119,7 @@ class _RegexFilter(RowFilter):
     """
 
     def __init__(self, regex):
-        self.regex = regex
+        self.regex = _to_bytes(regex)
```

Comment:
I was reacting as much to the changes propagated to the tests as to the code itself, which made me question the optimization:

- Do we  expect `to_pb` to be called frequently for a given filter?
- Is there a chance that `to_pb` might never get called for some filters?
- Is the cost of encoding inside `to_pb` if it were called more than once measurable?

Having raised the questions, I'll trust your sense about how the tradeoffs should be made.